app/module/drivers/sensor/battery/battery_common.c

```diff
@@ -29,15 +29,37 @@ int battery_channel_get(const struct battery_value *value, enum sensor_channel c
     return 0;
 }
 
-uint8_t lithium_ion_mv_to_pct(int16_t bat_mv) {
-    // Simple linear approximation of a battery based off adafruit's discharge graph:
-    // https://learn.adafruit.com/li-ion-and-lipoly-batteries/voltages
-
-    if (bat_mv >= 4200) {
-        return 100;
-    } else if (bat_mv <= 3450) {
-        return 0;
+uint8_t lithium_ion_mv_to_pct(int16_t batt_mv) {
+    // Lookup table of slope formulas for calculating remaining battery capacity.
+    //
+    // The current set of values acheives the same as: bat_mv * 2 / 15 - 459;
+    //
+    // Future updates sould include more complete lookup tables for different
+    // battery types slectable through config flags.
```

Comment:
A few typos. Also, a bit pedantic, but this is a lookup table of `{voltage, state_of_charge}` points, not a lookup table of formulas.

```suggestion
    // Lookup table for calculating remaining battery capacity.
    //
    // The current set of values achieves the same as: bat_mv * 2 / 15 - 459;
    //
    // Future updates should include more complete lookup tables for different
    // battery types selectable through config flags.
```